File: C/0003.c
#include <stdio.h>

// Function to find the largest prime factor
long largest_prime_factor(long n) {
    long largest = 0;
    
    // Remove all factors of 2
    while (n % 2 == 0) {
        largest = 2;
        n /= 2;
    }
    
    // Remove all factors of odd numbers
    for (long i = 3; i * i <= n; i += 2) {
        while (n % i == 0) {
            largest = i;
            n /= i;
        }
    }
    
    // If n is a prime number greater than 2
    if (n > 2) {
        largest = n;
    }
    
    return largest;
}

int main() {
    long number = 600851475143;
    long result = largest_prime_factor(number);
    
    printf("The largest prime factor of %ld is %ld\n", number, result);
    
    return 0;
}

File: R/0003.r
largest_prime_factor <- function(n) {
  max_prime <- -1

  # Remove all factors of 2
  while (n %% 2 == 0) {
    max_prime <- 2
    n <- n %/% 2
  }

  # Check odd numbers from 3 upwards
  i <- 3
  while (i * i <= n) {
    while (n %% i == 0) {
      max_prime <- i
      n <- n %/% i
    }
    i <- i + 2
  }

  # If remaining n is a prime number greater than 2
  if (n > 2) {
    max_prime <- n
  }

  return(max_prime)
}

number <- 600851475143
result <- largest_prime_factor(number)
cat("The largest prime factor of", number, "is", result, "\n")